src/twisted/web/http.py

```diff
@@ -2405,27 +2413,29 @@ def _finishRequestBody(self, data):
         self._dataBuffer.append(data)
         self.allContentReceived()
 
+    def _failChooseTransferDecoder(self) -> bool:
+        """
+        Utility to indicate failure to choose a decoder.
+        """
+        self._respondToBadRequestAndDisconnect()
+        self.length = None
+        return False
+
     def _maybeChooseTransferDecoder(self, header, data):
         """
         If the provided header is C{content-length} or
         C{transfer-encoding}, choose the appropriate decoder if any.
 
         Returns L{True} if the request can proceed and L{False} if not.
         """
-
-        def fail():
-            self._respondToBadRequestAndDisconnect()
-            self.length = None
-            return False
-
         # Can this header determine the length?
         if header == b"Content-Length":
             if not data.isdigit():
-                return fail()
+                return self._failChooseTransferDecoder()
             try:
                 length = int(data)
             except ValueError:
-                return fail()
+                return self._failChooseTransferDecoder()
```

Comment:
I think that this is now unreachable due to the `data.isdigit()` check above. The change snuck in because the `isdigit()` check was added as part of a [security advisory](https://github.com/twisted/twisted/security/advisories/GHSA-c2jg-hw38-jrqq) and that doesn't run the regular build checks.

Deleting this `try...except` seems the right option here.